### PR TITLE
FXIOS-1037 ⁃ Fix #7449 - Restoring last tab is broken, shows white screen

### LINKED ISSUE #7449: FXIOS-1009 ⁃ Restoring last tab is broken: white screen
## Steps to reproduce

1. Open a few tabs. Load pages.
2. Swipe Firefox away.
3. Reopen Firefox.

### Expected behavior

1. Last open tab loads.

### Actual behavior

1. Blank white screen, no URL in URL bar.
2. Actual title and page preview are visible for the affected tab in the tab switcher.
3. Switching to another tab will load that tab, but switching back never works.
4. Rather than a "Reload" button, the bottom bar shows a "Search" icon.

### Device & build information
* Device: iPhone X
* Build version: 2719

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1009)

### PR DESCRIPTION
**Pull request by nbhasin2**
I have added back an important step to migrate our internal urls without which we would have incorrect urls.

**Incorrect Internal Url Format**
Without our migration, internal urls will look something like below isn't right as this url should be very clean with proper ID appended to the end `[SOME UUID] and this wasn't happening as we were straight away saving these without cleaning internal url.

_internal://local/sessionrestore?history=%7B%22currentPage%22:0,%22history%22:%5B%22internal:%5C/%5C/local%5C/sessionrestore?url=internal%253A%252F%252Flocal%252Fabout%252Fhome%253Fuuidkey%253DF470F780~~C167~~41A9~~A712~~15A4F52C06A2%2523panel%253D0&uuidkey=[SOME UUID]%22,%22https:%5C/%5C/www.amazon.com%5C/%22%5D%7D_

**Correct Internal Url Format**
With migration (cleanup) we haveclean and proper url that we save and are able to load websites from.

_internal://local/sessionrestore?url=https%3A%2F%2Fwww.amazon.com%2F&uuidkey=[SOME UUID]_

**Whats still broken:**
Tabs widget seems to be partially broken but that is a different issue altogether.

Reference Old PR: https://github.com/mozilla-mobile/firefox-ios/pull/4387

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1037)

### CODE REVIEW

**Comment by nbhasin2** [comment]
Making it a draft until I fix the widget issue.

**Comment by nbhasin2** [comment]
Moving it out of draft and will fix any widget issue in separate PR

File: Client/Frontend/Browser/SessionData.swift [hunk 1]
```diff
@@ -6,6 +6,33 @@ import Foundation
 
 import Shared
 
+private func migrate(urls: [URL]) -> [URL] {
+    return urls.compactMap { url in
+        var url = url
+        let port = AppInfo.webserverPort
+        [("http://localhost:\(port)/errors/error.html?url=", "\(InternalURL.baseUrl)/\(SessionRestoreHandler.path)?url=")
+            // TODO: handle reader pages ("http://localhost:6571/reader-mode/page?url=", "\(InternalScheme.url)/\(ReaderModeHandler.path)?url=")
```
Comment (boek) [inline Client/Frontend/Browser/SessionData.swift:14]: Is this something that we'll want to fix for v29? Should we file a follow up?

**Review by boek — APPROVED** [review]
The code looks good, though as someone without any context it might be nice to comment or have a test to document what `migrate` does 😀

File: Client/Frontend/Browser/SessionData.swift (hunk 1 above)
Comment (nbhasin2) [reply Client/Frontend/Browser/SessionData.swift:14]: This is more of an enhancement for our test suit. we can leave it for a later release.

**Comment by nbhasin2** [comment]
@boek Added a comment to further explain this code and where it originated from along with a commit link. Hoping this helps in further understanding this code if someone else works on it in future. 

Ref old pr - https://github.com/mozilla-mobile/firefox-ios/pull/4387

**Comment by boek** [comment]
:shipit:

**Pull request merged**